### DISCUSSION TITLE
JIT Classes Problems

### DISCUSSION THREAD

**Discussion by mehdad-hussain** [tailwindlabs/tailwindcss#4973] [Help] [answered]
I am using JIT in a react project.
But few classes not generating in JIT mode !!
```javascript
<input
        className='w-full  border-2 border-blue-500 '
        type={props.type}
        placeholder={props.placeholder}
        id={props.id}
        onChange={changeHandler}
        value={inputState.value}
      />
```
This is my initial code with Normal mode in Tailwind config file. 
```javascript
<input
        className='w-3/4 p-2  border-2 border-red-500 '
        type={props.type}
        placeholder={props.placeholder}
        id={props.id}
        onChange={changeHandler}
        value={inputState.value}
      />
```
In normal mode when I am changing classes like width border-radius ,or border-color. It's working.
But in JIT mode only border-radius & width class change taking effect. But border-color not changing.
It is going back to browser default style. Is border-color class is not created in build process of JIT mode ?
If I change border-color in normal mode first, then it's also working in JIT mode !!! when I enable JIT mode again after disabling it. 
But if I change border-color again like ex- `border-pink-500` then it's not working again in JIT mode.
Help me please, where I am making mistake :( ?

**Comment by droplab** [comment, marked as answer]
FWIW, I've run into arbitrary rules not being created at times as well. I usually stop the dev server, run a `yarn build` start the dev server again and the rules are created. Hope this helps!

**Reply by mehdad-hussain** [reply]
Thanks, it's working, in my case `npm build`. But why I have to stop and start dev server again ? Is this a issue of JIT mode (as normal mode doesn't need dev server to be restated)?
Can you please give me some more elaborate explanation on  "arbitrary rules not being created"?
Sorry for asking, as I am new in Tailwind. And thanks again for your help.
